### PR TITLE
Marketplace init

### PR DESCRIPTION
**Pull request by ringaile**

### CODE REVIEW

File: web3/contracts/MirrorMarketplace.sol [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    struct Listing {
+        IERC721 token;
+        address seller;
+        uint tokenId;
+        uint price;
+        uint timestamp;
+        ListingStatus status;
+    }
+
+    mapping (uint => Listing) private listings;
```
Comment (jenpaff) [inline web3/contracts/MirrorMarketplace.sol:24]: whats the mapping for?

File: web3/contracts/MirrorMarketplace.sol [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        Listed,
+        Cancelled,
+        Sold
+    }
+
+    struct Listing {
+        IERC721 token;
+        address seller;
+        uint tokenId;
+        uint price;
```
Comment (jenpaff) [inline web3/contracts/MirrorMarketplace.sol:19]: so I was going to say shouldn't be this a floating number, but seems like there's no floating numbers in solidity? :D 
same as the tokenId/listingId I would've supposed to use a uuid of some sort, but seems like solidity is also restricted there

Comment (ringaile) [reply web3/contracts/MirrorMarketplace.sol:19]: yes there is no floating numbers in solidity. but we set a price in gwei so i do not think we need a floating number

File: web3/contracts/MirrorMarketplace.sol (hunk 1 above)
Comment (ringaile) [reply web3/contracts/MirrorMarketplace.sol:24]: so mapping is data structure in solidity. 
we need to have a list of listings to add and modify our listings.

solidity has array and mapping data structures to store lists

File: web3/contracts/MirrorMarketplace.sol (hunk 2 above)
Comment (jenpaff) [reply web3/contracts/MirrorMarketplace.sol:19]: ok makes sense !

**Review by jenpaff — APPROVED** [review]
great looks good to me :)

**Pull request merged**